CareKitStore/CareKitStore/Structs/OCKScheduleElement.swift

```diff
@@ -88,8 +88,6 @@ public struct OCKScheduleElement: Codable, Equatable {
         }
     }
 
-    private let calendar: Calendar
```

Comment:
#689 adds `calendar` to be a required encoded/decoded property which will break any `OCKTask` that's currently stored on a server in JSON because previous versions of CareKit never contained this property.

~Making the property a `var` and setting a default value will allow older saved OCKTasks's to be decoded as well as new ones (preserving all new updates with backwards compatibility).~ The other way would be to write a custom decoder for `OCKSheduleElement`, but that's more work and I don't think it's necessary.

Decoding Calendar.current after a few minutes seemed to cause crashes after ~10 minutes when `makeValue` is called during the hydration of a OCKTask from CoreData. Checking to see if this latest commit fixes the crash.

Reply:
Nice catch. I wonder if a good solution here is to pass in the calendar when it's needed and store it elsewhere (maybe the OCKStore?).

> OCKScheduleElement after a few minutes seems to cause crashes after ~10 minutes when makeValue is called during the hydration of a OCKTask from CoreData.

Can you expand on that a bit? Curious what you discovered and if the issue is caused by this stored calendar?

Reply:
I'm looking further into this as I know how to replicate it, but if I set breakpoints and step through them. This leads to believe that it could be related to how I'm setting up the Stores or there could be a threading issue with the streaming store. I only get the crash on Xcode 14.3, it doesn't happen on Xcode 14.2

Reply:
This is where I get the crash the most in Xcode 14.3 which is why I originally thought it had something to do with `Calendar.current`: 
<img width="1266" alt="image" src="https://user-images.githubusercontent.com/8621344/231296698-5eb139a4-3c81-4ec3-a5d4-bef3001778ce.png">

The call stack shows some of the streaming going on. In addition, this is in the console window:

```bash
2023-04-11 17:53:16.229604-0400 OCKSample[50867:4961092] [CareKit] Beginning to stream data from the store
2023-04-11 17:53:16.259117-0400 OCKSample[50867:4961092] [CareKit] Beginning to stream data from the store
2023-04-11 17:53:16.592551-0400 OCKSample[50867:4961396] [error] error: NULL _cd_rawData but the object is not being turned into a fault
CoreData: error: NULL _cd_rawData but the object is not being turned into a fault
(lldb) 
```

If I set a breakpoint here, `po` a couple properties, then run, it won't crash

Reply:
> Nice catch. I wonder if a good solution here is to pass in the calendar when it's needed and store it elsewhere (maybe the OCKStore?).

This sounds like the right solution to me as I don’t see a reason to send it to the server